x-pack/legacy/plugins/ml/public/application/contexts/kibana/kibana_context.ts

```diff
@@ -21,8 +20,8 @@ export interface KibanaConfigTypeFix extends KibanaConfig {
 
 export interface KibanaContextValue {
   combinedQuery: any;
-  currentIndexPattern: IndexPattern;
-  currentSavedSearch: SavedSearch;
+  currentIndexPattern: IndexPattern; // TODO this should be IndexPattern or null
```

Comment:
Is this TODO for a followup?

Reply:
The plan is to use kibana's react utils which has its own kibanaContext.
We need to investigate whether it will cover all our needs, hopefully it is extendable so we can add items like `currentIndexPattern` and `currentSavedSearch` which a lot of our pages rely on.